docs/src/main/asciidoc/extension-codestart.adoc

```diff
@@ -0,0 +1,259 @@
+
+== Where are the Quarkus Extension Codestarts located
+
+- In Quarkus core repo, the extension codestarts are all in the same https://github.com/quarkusio/quarkus/tree/main/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts[module, window="_blank"].
+
+- RESTEasy, RESTEasy Reactive and Spring Web extension codestarts are part of https://github.com/quarkusio/quarkus/tree/main/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts[the base codestarts, window="_blank"].
+
+- For non-core extensions, the codestart is located in the runtime module (with special instruction in the pom.xml to generate a separated codestart artifact).
+
+== Base codestarts
+
+The https://github.com/quarkusio/quarkus/tree/main/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus[base codestarts, window="_blank"] contains templates to create project, buildtool, languages, config & tooling files.
+
+== Writing an Extension Codestart
+
+As we discussed before, the base project files (pom.xml, dockerfiles, ...) are already generated by some existing base codestarts. Thanks to this, we can only focus on the important, the starter code for the extension.
+
+The codestart should not have any business code, instead, it should contain some stub data/hello world to compile. The idea is to bring code that is the starting point to everyone using the extension.
+
+== Writing an Extension Codestart in Quarkus Core
+
+- Duplicate an existing https://github.com/quarkusio/quarkus/tree/main/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts[Quarkus core extension codestarts, window="_blank"]. `resteasy-qute-codestart` is a good base when the code will expose a web resource, in other cases, `config-yaml-codestart` is a better base. More info on the <<directory-structure>>.
+
+- Edit the <<codestart-yml>>:
+
+- Create the extension binding in the extension metadata (https://github.com/quarkusio/quarkus/blob/main/extensions/config-yaml/runtime/src/main/resources/META-INF/quarkus-extension.yaml#L12-L17[example, window="_blank"]). *Thanks to this, the codestart is added when the user selects the extension*
+
+- Add the readme <<readme-md>> section template.
+
+- Add the code in the language folder (it is recommended to at least provide java and kotlin). *You have to use org.acme as package name: <<org-acme-package>>*. It is possible to use <<qute-templates>> if needed.
+
+- Optionally, Add the `index.html` section template (<<index-html>>).
+
+- Optionally, add some resources (`./base` directory if they are non language specific)
+
+- Optionally, add the <<app-config>>.
+
+- Create an <<integration-test>> *You can use it to get your codestart generated as a project*.
+
+You can also use it for dev using the `buildAllProjectsForLocalUse` (It's disabled by default because already covered by `QuarkusCodestartBuildIT`).
```

Comment:
```suggestion
You can also use it for dev using the `buildAllProjectsForLocalUse` (Disabled by default because it's already covered by `QuarkusCodestartBuildIT`).
```